ls1intum/Artemis
# `Iris`: Allow only admins to change certain settings

> [!NOTE]
> Only deploy on TS9




### Checklist
#### General


#### Server

### Motivation and Context


We want to prevent just any instructor from being able to enable Iris.

### Description

This PR adds a quick fix for this. This is only temporary until the settings system is reworked.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Admin (your TUM account)
- TS9

1. Log in to Artemis as Instructor
2. Navigate to a course management page
3. Open the Iris settings
4. Change the enabled status & the preferred model
5. Save
6. The values should be reverted
7. Go to an exercise management page
8. Do the same with the same result
9. Log in to Artemis as Admin
10. Repeat 2-8, but this time the settings actually save

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage






Unchanged
